packages/apollo-gateway/src/QueryPlan.ts

```diff
@@ -38,20 +38,73 @@ export interface ParallelNode {
 export interface FetchNode {
   kind: 'Fetch';
   serviceName: string;
-  selectionSet: SelectionSetNode;
-  variableUsages?: { [name: string]: VariableDefinitionNode };
-  requires?: SelectionSetNode;
-  internalFragments: Set<FragmentDefinitionNode>;
-  source: string;
+  variableUsages?: string[];
+  requires?: SelectionNode[];
+  operation: string;
 }
+
 export interface FlattenNode {
   kind: 'Flatten';
   path: ResponsePath;
   node: PlanNode;
 }
 
+export type SelectionNode = FieldNode | InlineFragmentNode;
+
+export interface FieldNode {
+  readonly kind: 'Field';
+  readonly alias?: string;
+  readonly name: string;
+  readonly selections?: SelectionNode[];
+}
+
+export interface InlineFragmentNode {
+  readonly kind: 'InlineFragment';
+  readonly typeCondition?: string;
+  readonly selections: SelectionNode[];
+}
+
 export function serializeQueryPlan(queryPlan: QueryPlan) {
   return prettyFormat(queryPlan, {
     plugins: [queryPlanSerializer, astSerializer],
   });
 }
+
+export function getResponseName(node: FieldNode): string {
+  return node.alias ? node.alias : node.name;
+}
+
+/**
+ * Converts a GraphQL-js SelectionNode to our newly defined SelectionNode
+ *
+ * This function is used to remove the unneeded pieces of a SelectionSet's
+ * `.selections`. It is only ever called on a query plan's `requires` field,
+ * so we can guarantee there won't be any FragmentSpreads passed in. That's why
+ * we can ignore the case where `selection.kind === Kind.FRAGMENT_SPREAD`
+ */
+export const trimSelectionNodes = (
+  selections: readonly GraphQLJSSelectionNode[],
+): SelectionNode[] => {
+  let remapped: SelectionNode[] = [];
```

Comment:
Could this be a `const` (If not, maybe a comment to explain, otherwise it doesn't look to be immediately clear to me how the re-assignment is happening).

Reply:
I think I originally did this as a signal that `remapped` was being mutated, at least in my mind, but there's no _need_ for the `const`. I swapped it out :)